go/vt/vtgate/vindexes/vschema.go

```diff
@@ -245,6 +248,116 @@ func buildKeyspaces(source *vschemapb.SrvVSchema, vschema *VSchema) {
 		vschema.Keyspaces[ksname] = ksvschema
 		ksvschema.Error = buildTables(ks, vschema, ksvschema)
 	}
+
+	// In the second loop, build references.
+	for ksname, ks := range source.Keyspaces {
+		ksvschema := vschema.Keyspaces[ksname]
+		if err := buildReferences(ks, vschema, ksvschema, allowUnresolvedReferences); err != nil && ksvschema.Error == nil {
+			ksvschema.Error = err
+		}
+	}
+
+	// In the third loop, build global tables.
+	for ksname, ks := range source.Keyspaces {
+		ksvschema := vschema.Keyspaces[ksname]
+		buildGlobalTables(ks, vschema, ksvschema)
+	}
+}
+
+func buildGlobalTables(ks *vschemapb.Keyspace, vschema *VSchema, ksvschema *KeyspaceSchema) {
+	for tname, t := range ksvschema.Tables {
+		// If the keyspace requires explicit routing, don't include any of
+		// its tables in global tables.
+		if !ks.RequireExplicitRouting {
```

Comment:
I think go wants you to inverse this and continue:

```golang
if ks.RequireExplicitRouting {
  continue
}
```

Reply:
This should also move out of the for loop at the top of the method